Code-the-Dream-School/cc-prac-team4-front
# Pp 68 create donate page

_Alona(functionality) @AlonaVladymyrovaTrinity , Liubov(Cascading Style Sheets) @LiubovCass , Vitalii(Stripe CheckoutForm) @vitaliipp_

**Donate page created**

**Pull Request Description:**

This pull request contains changes in the next main following files: Donate.js, and DonateProcess.js, CheckoutForm.js.

/src/component/Donate/Donate.js: This file contains a React component that represents the donation page. It includes a form for selecting a donation amount and handles the validation of the custom amount. It also sets cookies with the selected donation amount and custom amount and redirects the user to the donation process page.

/src/component/Donate/DonateProcess.js: This file contains a React component that represents the donation process page. It retrieves the client secret for the payment intent from the server based on the donation amount, and then renders the CheckoutForm component with the appropriate options.

/src/component/Donate/CheckoutForm.js: This file contains a React component that represents the checkout form for making a donation. It handles the Stripe payment integration and displays the payment element and relevant messages based on the payment status.

**Please ensure that the necessary environment variables and dependencies are properly set up before testing the pull request.(Before checking this pull request ask us about secret keys.)**

**Testing Steps:**

- Install the required dependencies using npm install.
Install:

```
npm install 
npm install @stripe/react-stripe-js @stripe/stripe-js
npm install react-icons
```

- **Make sure the Stripe API keys are properly configured in the environment variables.**
- Run the application locally using npm start.
- Open your browser and access the donation page.

[http://localhost:3000/donate](http://localhost:3000/donate)

- Select a donation amount or enter a custom amount.
- Submit the donation form and verify that the validation messages are displayed correctly for invalid inputs.
- If the input is valid, the donation process page should be loaded.

[http://localhost:3000/process/donate](http://localhost:3000/process/donate)

- Verify that the checkout form is displayed and functional.
- Test the Stripe integration by entering valid **testing payment details** and completing the donation.

> **Testing payment details:**
> Card number: 
> 4242 4242 4242 4242
> Expiration: any(but date should be in future)
> CVC: any
> Country: United States
> Zip: any

- Verify that the payment status message is displayed correctly based on the outcome.
- Test different scenarios, such as successful payment, processing payment, and failed payment, to ensure proper handling of different payment statuses.
- Confirm that the cookies are set correctly with the selected donation amount and custom amount.
- Test the overall flow of the donation process from start to finish, ensuring that the data is accurately processed and the user is redirected appropriately.

<img width="1432" alt="Screenshot 2023-07-02 at 1 53 26 AM" src="https://github.com/Code-the-Dream-School/cc-prac-team4-front/assets/2952900/6fb650ea-162f-4b48-93ee-79a1c55b0a25">
<img width="1430" alt="Screenshot 2023-07-02 at 1 54 17 AM" src="https://github.com/Code-the-Dream-School/cc-prac-team4-front/assets/2952900/04c531bf-b3a8-449a-88b9-5e56cdbff4b3">
